code/datums/components/storage/concrete/fish_case.dm

```diff
@@ -0,0 +1,19 @@
+/datum/component/storage/concrete/fish_case
+	max_items = 1
+	can_hold_description = "fish and aquarium equipment"
+
+/datum/component/storage/concrete/fish_case/can_be_inserted(obj/item/I, stop_messages, mob/M)
+	. = ..()
+	/// Activate deferred components if any.
+	SEND_SIGNAL(I,COMSIG_AQUARIUM_BEFORE_INSERT_CHECK)
```

Comment:
this signal is a proc, you should find another way to do this that isn't this. (I wouldn't be able to tell you specifically how because storage code is wack)